sharing/data_share.go

```diff
@@ -3,20 +3,28 @@ package sharing
 import (
 	"context"
 
+	"github.com/databricks/databricks-sdk-go/service/sharing"
 	"github.com/databricks/terraform-provider-databricks/common"
 )
 
 func DataSourceShare() common.Resource {
 	type ShareDetail struct {
-		Name      string             `json:"name,omitempty" tf:"computed"`
-		Objects   []SharedDataObject `json:"objects,omitempty" tf:"computed,slice_set,alias:object"`
-		CreatedAt int64              `json:"created_at,omitempty" tf:"computed"`
-		CreatedBy string             `json:"created_by,omitempty" tf:"computed"`
+		Name      string                     `json:"name,omitempty" tf:"computed"`
+		Objects   []sharing.SharedDataObject `json:"objects,omitempty" tf:"computed,slice_set,alias:object"`
```

Comment:
Some changes would be required before we can do this, for example: 
`StartVersion` in `SharedDataObject` is annotated `tf:"suppress_diff" ` whereas it is not in Go SDK struct.

Also, DataObjectType is string in `SharedDataObject` but SharedDataObjectDataObjectType in Go SDK which I think is okay but needs to be verified if this doesn't cause any other issue (eg: if its used in test, then that would need to be upgraded, if it's not in test, might be good to add a test).

Reply:
👍 addressed in 20b08cbbb0ba0cbbff333aa72deba511ec81c637